server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java

```diff
@@ -737,6 +737,13 @@ public static ClientRepresentation toRepresentation(ClientModel clientModel, Key
         rep.setNodeReRegistrationTimeout(clientModel.getNodeReRegistrationTimeout());
         rep.setClientAuthenticatorType(clientModel.getClientAuthenticatorType());
 
+        String clientId = clientModel.getClientId();
+        final String realmClientSuffix = "-realm";
+        //check if client is realm client
+        if(clientId != null && clientId.endsWith(realmClientSuffix) && session.realms().getRealmByName(clientId.substring(0, clientId.length() - realmClientSuffix.length())) != null) {
```

Comment:
The clients like `master-realm` or `test-realm` are supposed to be present just in the `master` realm. Maybe another condition for `&& (Config.getAdminRealm().equals(clientModel.getRealm().getName()` should be added here? Other clients have client called `realm-management` for handle admin REST permissions.

Reply:
+1 for handle the `broker` and `realm-management` clients the same way like `test-realm` or `master-realm` clients.

For other clients like `security-admin-console` or `admin-cli`, I am not sure as those are "real" client applications to which user can login and can issue token. So it can still makes sense like being able to show tabs like `Keys` or `Credentials` for these clients (although those are both public clients and hence it won't be shown by default, but at least for `admin-cli` it makes perfect sense to switch it to confidential and then change credentials or other things..).